pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -305,37 +305,68 @@ type VitessShardStatus struct {
 	// Cells is a list of cells in which any tablets for this shard are deployed.
 	Cells []string `json:"cells,omitempty"`
 
-	// HasMaster is a condition indicating whether the Vitess topology
-	// reflects a master for this shard.
-	HasMaster corev1.ConditionStatus `json:"hasMaster,omitempty"`
-	// HasInitialBackup is a condition indicating whether the initial backup
-	// has been seeded for the shard.
-	HasInitialBackup corev1.ConditionStatus `json:"hasInitialBackup,omitempty"`
+	// Conditions is a map of all VitessShard specific conditions we want to set and monitor.
+	Conditions map[VitessShardConditionType]*VitessShardCondition `json:"conditions,omitempty"`
```

Comment:
Status conditions are conventionally a list rather than a map. Maybe a map is better but we should think about whether it's worthwhile to be conventional, and also whether there is a reason the convention avoids maps.

Reply:
We chatted about this offline, and it sounds like we're sticking to a map because it's a better choice and worth breaking with convention?

Reply:
We'll do it live.